### PR TITLE
[fix](cluster key) fix cluster key duplicated key

### PR DESCRIPTION
**Pull request by mymeiyi**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
the duplicated keys is happend when:
1. do schema change and not finished
2. write data and the publish phase skip calculate delete bitmap
3. compact the newly rowsets without delete bitmap: the mow with cluster key need to do compaction with delete bitmap

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by bobhan1 — APPROVED** [review]

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run external

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -872,6 +875,60 @@ void Compaction::construct_index_compaction_columns(RowsetWriterContext& ctx) {
...
+    {
+        std::shared_lock meta_rlock(_tablet->get_header_lock());
+        if (_tablet->tablet_state() != TABLET_NOTREADY) {
+            return Status::OK();
+        }
+    }
+    OlapStopWatch watch;
+    std::vector<RowsetSharedPtr> rowsets;
+    for (const auto& rowset : _input_rowsets) {
+        std::lock_guard rwlock(tablet()->get_rowset_update_lock());
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp:890]: it's not necessary to hold the rowset_update_lock

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -872,6 +875,60 @@ void Compaction::construct_index_compaction_columns(RowsetWriterContext& ctx) {
...
+    {
+        std::shared_lock meta_rlock(_tablet->get_header_lock());
+        if (_tablet->tablet_state() != TABLET_NOTREADY) {
+            return Status::OK();
+        }
+    }
+    OlapStopWatch watch;
+    std::vector<RowsetSharedPtr> rowsets;
+    for (const auto& rowset : _input_rowsets) {
+        Status st = _tablet->update_delete_bitmap_without_lock(_tablet, rowset, &rowsets);
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp:918]: need to hold the header lock?

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**